doc/source/specification.rst

```diff
@@ -1,104 +1,112 @@
 Specification
 =============
-
-Description of how MorphIO sees morphologies.
-
-Formats:
---------
-MorphIO supports ASC (aka Neurolucida), H5 and SWC extensions
+Description of how MorphIO sees morphologies. MorphIO supports :ref:`.asc <specification-neurolucida>` (Neurolucida),
+`.h5`_ (H5 version 1) and `.swc`_ extensions.
 
 Soma formats:
 -------------
-(For more information, see: `Soma Format <http://neuromorpho.org/SomaFormat.html>`_\ ) MorphIO
-recognizes several kinds of soma format. The soma type is determined according to the morphology
-type and the number of data points:
+MorphIO implements Soma specification from `NeuroMorpho`_, and recognizes several kinds of soma format.
 
-Undefined soma SOMA_UNDEFINED
-*****************************
+No Soma
+*******
+MorphIO can read correctly a file without a soma. The soma type of such file will be ``SOMA_UNDEFINED``.
 
-MorphIO can read correctly a file without a soma. The latter will be of type "undefined".
-Additionnaly, a soma made of 2 points will be of type "undefined" as well. Indeed a 2-points soma
-does not make any sense in any of the other soma formats.
+Single point Soma
+*****************
 
-Soma single point SOMA_SINGLE_POINT
-***********************************
+.. image:: images/swc/sphere.svg
+   :scale: 100 %
+   :alt: soma single point
 
-If a soma is made of only one point it will be treated as a "single point soma" for H5 and ASC formats.
-The single point soma is represented as a sphere whose coordinate and radius are given by the soma point.
-SWC interprets such soma differently.
+If a soma is composed of only one point, it will be treated as a sphere with the center in this point, and the radius
+defined for this point. The type of such soma is ``SOMA_SINGLE_POINT``.
 
-Morphology file dependent soma type:
-************************************
+Two point Soma
+**************
+Only SWC is allowed to have two point Soma.
 
 SWC
 ^^^
 
-Soma single point
-~~~~~~~~~~~~~~~~~
-.. image:: images/swc/single.svg
-   :scale: 100 %
-   :alt: soma single point
-
-For H5 and ASC format this single point denotes a sphere. For SWC the single point is interpreted as a cylinder. The
-area of this cylinder is equivalent to the area of a sphere as like this sphere was defined for H5 and ASC format.
-
-Soma two point
-~~~~~~~~~~~~~~
 .. image:: images/swc/two_pt.svg
    :scale: 100 %
    :alt: soma two point
 
 Soma is a cylinder with those two points as centers of the cylinder's bases. A point in this case defines a circle for
-the cylinder's base.
+the cylinder's base. The type of such soma is ``SOMA_CYLINDERS``.
+
 
-Soma three point Neuromorpho
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-.. image:: images/swc/neuromorpho.svg
+Three point Soma
+****************
+Interpretation of three point soma differs among formats.
```

Comment:
This conflicts with http://neuromorpho.org/SomaFormat.html. There it does not say that only SWC must implement this standard. There is no explicit mentioning of SWC vs H5 vs ASC. Should we treat such somas the same in all three formats ASC, H5, SWC?

Reply:
In these specs, only the swc is mentioned. So I am not sure we can propagated the behavior to any other formats. Or we should change the specs from the h5/asc...

Reply:
Well, the spec is vague. I don't see that it explicitly says it is only for SWC however it mentions it the most. H5 not at all, and there are 2 examples of ASC.

Reply:
Remember that H5/ASC are defining *contours*, and SWC is stacked cylinders - thus, three points in h5/asc makes a (strange, but valid) triangular contour.